core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -386,6 +386,16 @@ private static String formatFragment(
         return builder.toString();
     }
 
+    private static TypeProvider getTypeProvider(List<PlanFragment> fragments)
+    {
+        // somewhat faster than java stream
+        Map<Symbol, Type> symbols = new HashMap<>();
+        for (PlanFragment fragment : fragments) {
+            fragment.getSymbols().forEach(symbols::put);
```

Comment:
previous code used `distinct` on entries, so any duplicates where "coherent"

Here, you overwrite entries in `symbols` without checking that symbol types do agree.
Please verify that.
Best way to do it is by using `toImmutableMap` collector or `ImmutableMap.builder()`.